back/app/services/project_copy_service.rb

```diff
@@ -78,44 +77,31 @@ def export(
   private
 
   def yml_content_builder_layouts(shift_timestamps: 0)
-    layout_images_mapping = {}
-
-    layouts = ContentBuilder::Layout.where(content_buildable_id: @project.id).map do |layout|
-      craftjs = layout.craftjs_jsonmultiloc
-
-      craftjs.each_key do |locale|
-        craftjs[locale].each_value do |node|
-          next unless ContentBuilder::LayoutService.new.craftjs_element_of_type?(node, 'Image')
-
-          source_image_code = node['props']['dataCode']
-          new_image_code = ContentBuilder::LayoutImage.generate_code
-          node['props']['dataCode'] = new_image_code
-          layout_images_mapping[source_image_code] = new_image_code
-        end
-      end
-
+    ContentBuilder::Layout.where(content_buildable_id: @project.id).map do |layout|
       yml_layout = {
         'content_buildable_ref' => lookup_ref(layout.content_buildable_id, :project),
         'content_buildable_type' => layout.content_buildable_type,
         'code' => layout.code,
         'enabled' => layout.enabled,
-        'craftjs_jsonmultiloc' => craftjs,
+        'craftjs_jsonmultiloc' => layout.craftjs_jsonmultiloc,
         'created_at' => shift_timestamp(layout.created_at, shift_timestamps)&.iso8601,
         'updated_at' => shift_timestamp(layout.updated_at, shift_timestamps)&.iso8601
       }
       store_ref yml_layout, layout.id, :content_builder_layout
       yml_layout
     end
-
-    [layouts, layout_images_mapping]
   end
 
-  def yml_content_builder_layout_images(layout_images_mapping, shift_timestamps: 0)
-    source_image_codes = layout_images_mapping.keys.map(&:to_s)
+  def yml_content_builder_layout_images(shift_timestamps: 0)
+    source_image_codes = []
+
+    ContentBuilder::Layout.where(content_buildable_id: @project.id).map do |layout|
+      source_image_codes += ContentBuilder::LayoutService.new.images(layout).pluck(:code)
+    end
```

Comment:
It could be argued that iterating over the source project's layouts again (first time is in `yml_content_builder_layouts`) will impact performance, but in reality I think we are talking a few milliseconds, which may well be saved anyway by using a less complex mechanism to copy the layouts.